examples/3.routes/8.tolls-and-ferries/map.js

```diff
@@ -0,0 +1,179 @@
+import mapboxgl from 'mapbox-gl';
+import { getDurationString } from '../../../utils';
+import * as mapboxPolyline from '@mapbox/polyline';
+
+mapboxgl.accessToken = 'pk.eyJ1IjoiY2hhcmdldHJpcCIsImEiOiJjazhpaG8ydTIwNWNpM21ud29xeXc2amhlIn0.rGKgR3JfG9Z5dCWjUI_oGA';
+
+const map = new mapboxgl.Map({
+  cooperativeGestures: true,
+  container: 'map',
+  style: 'mapbox://styles/chargetrip/ckgcbf3kz0h8819qki8uwhe0k',
+  zoom: 6,
+  center: [9.647487, 58.389399],
+});
+
+/**
+ * Draw a route on a map.
+ *
+ * Route object contains `polyline` data -  the polyline encoded route (series of coordinates as a single string).
+ * We need to decode this information first. We use Mapbox polyline tool (https://www.npmjs.com/package/@mapbox/polyline) for this.
+ * As a result of decoding we get pairs [latitude, longitude].
+ * To draw a route on a map we use Mapbox GL JS. This tool uses the format [longitude,latitude],
+ * so we have to reverse every pair.
+ *
+ * @param data {object} route specification
+ */
+export const drawRoutePolyline = data => {
+  const decodedData = mapboxPolyline.decode(data.polyline);
+  const reversed = decodedData.map(item => item.reverse());
+
+  drawRoute(reversed, data.legs);
+};
+
+/**
+ * Draw route polyline and show charging stations on the map.
+ *
+ * @param coordinates {array} Array of coordinates
+ * @param legs {array} route legs (stops) - each leg represents either a charging station, or via point or final point
+ */
+const drawRoute = (coordinates, legs) => {
+  if (map.loaded()) {
+    drawPolyline(coordinates);
+    drawChargingTimes(legs);
+    showLegs(legs);
+  } else {
+    map.on('load', () => {
+      drawPolyline(coordinates);
+      drawChargingTimes(legs);
+      showLegs(legs);
+    });
+  }
+};
+
+/**
+ * Render the charging times at each station directly on top of it's marker.
+ * @param {array} legs - each leg represents either a charging station, or a via point or final point
+ */
+const drawChargingTimes = legs => {
+  legs.forEach((leg, idx) => {
+    if (idx == legs.length - 1) {
+      return;
+    }
+
+    const chargeTime = leg.chargeTime;
+    const hrs = ~~(chargeTime / 3600);
+    const mins = ~~((chargeTime % 3600) / 60);
```

Comment:
You can use the getDurationString function from the utils.js file. This is also used in the alternative routes example for instance.

Reply:
this is for the duration of charging on the map, not for the journey overview, where just the minutes are shown as necessary on the example design, same as in the routes example

Reply:
Might still want to extend the utils function and clean that one up. Maybe return a custom object that contains different types of formatting.